src/app/src/pages/ResetPassword.js

```diff
@@ -0,0 +1,140 @@
+        var passwordErrorSeen = false;
+
+        for (const [field, errorArray] of Object.entries(apiErrorData)) {
+            if (['uid', 'token'].includes(field)) {
+                errorDetail = 'The link you followed is no longer valid.';
+            } else if (!passwordErrorSeen) {
+                // Avoid duplicating password error messages
+                var formattedField;
+                if (field.includes('password')) {
+                    // new_password_1 -> Password
+                    formattedField = 'Password';
+                    passwordErrorSeen = true;
+                } else {
+                    formattedField = `${field[0].toUpperCase()}${field.slice(
+                        1
+                    )}`;
+                }
+                errorDetail = `${formattedField}: ${errorArray.join(' ')}`;
+            }
+        }
+        return errorDetail;
+    };
+
+    const forgotPasswordRequest = () => {
+        API.post(API_URLS.RESET, {
+            uid,
+            token,
+            new_password1: newPassword1,
+            new_password2: newPassword2,
+        })
+            .then(() => {
+                setSuccess(true);
+                setErrorDetail('');
+            })
+            .catch(apiError => {
+                setSuccess(false);
+                setErrorDetail(formatApiError(apiError.response?.data));
+            });
+    };
+
+    function makePasswordInput(setter, placeholder) {
+        return (
+            <Input
+                htmlSize={32}
+                width='auto'
+                bg='white'
+                aria-label={placeholder}
+                placeholder={placeholder}
+                type='password'
+                isInvalid={!!errorDetail && !disableSubmit}
+                disabled={disableSubmit}
+                onChange={({ target: { value } }) => setter(value)}
+                onKeyPress={e => {
+                    if (e.key === 'Enter') {
+                        forgotPasswordRequest(
+                            uid,
+                            token,
+                            newPassword1,
+                            newPassword2
+                        );
+                    }
+                }}
+            />
+        );
+    }
+
+    return (
+        <VStack bg='gray.100' h='100vh' spacing={5}>
+            <Box h='150px'></Box>
+            <Text fontFamily='heading' fontSize='lg'>
+                Reset Password
+            </Text>
+            <Box>
+                {errorDetail && (
+                    <Text textStyle='apiError' align='center'>
+                        {errorDetail}
+                    </Text>
+                )}
+                <Text fontFamily='body' fontSize='md' align='center'>
+                    {success
+                        ? 'Your password has been reset.'
+                        : disableSubmit
+                        ? ''
+                        : 'Choose a new password (8+ characters)'}
```

Comment:
Nested ternary expressions are difficult to read. An explicit `if`, while more verbose, would also be more readable.